### PR TITLE
TD-4345 step 6 updates

### PR DESCRIPTION
**Pull request by OluwatobiAwe**
Implemented the Step 6 process of the DSP Integration recommendation: external system checks employee’s identity

image

### CODE REVIEW

File: LearningHub.Nhs.UserProfileUI/Controllers/DigitalStaffPassportController.cs [hunk 1]
```diff
@@ -132,8 +173,10 @@ public async Task<IActionResult> DspReturn(string code, string state, string err
             }
             else if (state == "verifyCreation")
             {
-                var claims = await this.digitalStaffPassportService.GetCredentialDetails(code);
-                return this.View("ShowCredentialClaims", claims);
+                return await this.PersistIdentityVerification(code);
+
+                // var claims = await this.digitalStaffPassportService.GetCredentialDetails(code);
```
Comment (ColinBeebyHEE) [inline LearningHub.Nhs.UserProfileUI/Controllers/DigitalStaffPassportController.cs]: Do we need these lines here if they are commented out.

File: LearningHub.Nhs.UserProfileUI/Controllers/DigitalStaffPassportController.cs [hunk 2]
```diff
@@ -181,9 +235,35 @@ public IActionResult VerifyCredential(string credentialType)
...
+            {
+                var firstName = claims.FirstOrDefault(x => x.Type == "Identity.ID-LegalFirstName")?.Value;
+                var lastName = claims.FirstOrDefault(x => x.Type == "Identity.ID-LegalSurname")?.Value;
+
+                var userDetails = await this.userService.GetElfhUserByUserIdAsync(this.CurrentUserId);
+                if (firstName?.ToString().ToUpper() == userDetails.FirstName.ToUpper() &&
+                    lastName?.ToString().ToUpper() == userDetails.LastName.ToUpper())
+                {
+                    var uniqueIdentifier = claims.FirstOrDefault(x => x.Type == "Identity.UniqueIdentifier");
+                    if (uniqueIdentifier != null)
```
Comment (ColinBeebyHEE) [inline LearningHub.Nhs.UserProfileUI/Controllers/DigitalStaffPassportController.cs:248]: What should happen if uniqueIdentifier is null here. Would this indicate invalid identity verification?

File: LearningHub.Nhs.UserProfileUI/Services/DigitalStaffPassportService.cs [hunk 3]
```diff
@@ -292,7 +295,8 @@ private Dictionary<string, string> PopulateCertificateClaims(VerifiableCredentia
             ////claims.Add($"{verifiableCredential.ClaimPrefix}-Pedigree", "notset");
 
             // TODO - this hard-coded value needs to be picked up from the user's specific identity value
-            claims.Add("UniqueIdentifier", "3ed5a980-3a1b-45a0-9c50-6a58127cb19f");
+           // claims.Add("UniqueIdentifier", "3ed5a980-3a1b-45a0-9c50-6a58127cb19f");
```
Comment (ColinBeebyHEE) [inline LearningHub.Nhs.UserProfileUI/Services/DigitalStaffPassportService.cs]: Do we still need this commented line of code and the TODO?

File: LearningHub.Nhs.UserProfileUI/Controllers/DigitalStaffPassportController.cs (hunk 2 above)
Comment (OluwatobiAwe) [reply LearningHub.Nhs.UserProfileUI/Controllers/DigitalStaffPassportController.cs:248]: absolutely! DSP will always send "Identity.UniqueIdentifier" as part of their Identity verification response, and if it's absent, then it probably means the user hasn't completed the onboarding process at the DSP end of the process.

File: LearningHub.Nhs.UserProfileUI/Views/DigitalStaffPassport/ConfirmCredential.cshtml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            <dt class="nhsuk-summary-list__key">
+                <form asp-controller="DigitalStaffPassport" asp-action="SendCredential" method="post">
+                    @Html.HiddenFor(x=>x.VerifiableCredentialId)
+                    <button type="submit" class="nhsuk-button nhsuk-u-margin-bottom-0 nhsuk-u-margin-right-3">Confirm</button>
+                </form>
+            </dt>
+            <dd class="nhsuk-summary-list__value">
+                <a asp-action="Credentials" class="nhsuk-button nhsuk-button--secondary nhsuk-u-margin-bottom-0 nhsuk-u-padding-right-3">Cancel</a>
+            </dd>
+           @*  <dd class="nhsuk-summary-list__value">
```
Comment (ColinBeebyHEE) [inline LearningHub.Nhs.UserProfileUI/Views/DigitalStaffPassport/ConfirmCredential.cshtml]: Do we still need this commented out section?

File: LearningHub.Nhs.UserProfileUI/Services/DigitalStaffPassportService.cs (hunk 3 above)
Comment (OluwatobiAwe) [reply LearningHub.Nhs.UserProfileUI/Services/DigitalStaffPassportService.cs]: I cherry-picked my commit after updating the remote origin on my local PC. Obviously this approach seems problematic. I'll rework this PR

**Review by ColinBeebyHEE — COMMENTED** [review]
I've added a few comments to this PR - they are just queries rather then probems.

**Review by ColinBeebyHEE — APPROVED** [review]
Looks good to me.

**Pull request merged**